lib/engine/tile.rb

```diff
@@ -214,6 +214,83 @@ def inspect
       "<#{self.class.name}: #{name}, hex: #{@hex&.name}>"
     end
 
+    # returns array where the value at an index is which edge that city (based
+    # on index in self.cities) should be rendered
+    def edges_for_city_rendering
+      @_edges_for_city_rendering ||= {}
+      cached_val = @_edges_for_city_rendering[@rotation]
+      return cached_val unless cached_val.nil?
+
+      # array of hashes; index is city index, edges is nums of edges that city
+      # connects to
+      city_edges = cities.map do |city|
```

Comment:
this can just be
city_edges = Hash.new { |h, k| h[k] = [] }
paths.each  { |p| city_edges[p.city].concat(p.exits) if p.city } 

or something like this

why do you store an array of indices?

Reply:
I wanted to handle the edges always starting with the lowest edge that has a path. That edge could be connected to city 0 or city 1, so after sorting, city 1 could end up before city 0, but I needed to keep track of the original order of the cities.

I didn't thoroughly test this out, but my worry was that a tile could be rendered differently depending on which city was called city 0 and I wanted the tile rendering to be consistent regardless of which city was which. (I'm willing to admit worrying about this led to over-engineering :stuck_out_tongue: )

Will check out your code snippet now...

Reply:
well you can just sort by the min edge?

Reply:
Resolved in 6bcca6b and 41238cf